shared/chat/conversation/input-area/normal/location-popup.tsx

```diff
@@ -0,0 +1,142 @@
+import * as React from 'react'
+import * as Kb from '../../../../common-adapters'
+import * as Styles from '../../../../styles'
+import * as Container from '../../../../util/container'
+import * as RouteTreeGen from '../../../../actions/route-tree-gen'
+import * as Types from '../../../../constants/types/chat2'
+import * as Chat2Gen from '../../../../actions/chat2-gen'
+import * as Constants from '../../../../constants/chat2'
+import {isIOS} from '../../../../constants/platform'
+import {imgMaxHeightRaw, imgMaxWidthRaw} from '../../messages/attachment/image/image-render'
+import HiddenString from '../../../../util/hidden-string'
+
+type Props = Container.RouteProps<{conversationIDKey: Types.ConversationIDKey}>
+
+const LocationPopup = (props: Props) => {
+  // state
+  const conversationIDKey = Container.getRouteProps(props, 'conversationIDKey', Constants.noConversationIDKey)
+  const {httpSrvAddress, httpSrvToken, location} = Container.useSelector(state => ({
+    httpSrvAddress: state.config.httpSrvAddress,
+    httpSrvToken: state.config.httpSrvToken,
+    location: state.chat2.lastCoord,
+  }))
+  const [mapLoaded, setMapLoaded] = React.useState(false)
+  // dispatch
+  const dispatch = Container.useDispatch()
+  const onClose = () => {
+    dispatch(RouteTreeGen.createClearModals())
+  }
+  const onLocationShare = (duration: string) => {
+    onClose()
+    dispatch(
+      Chat2Gen.createMessageSend({
+        conversationIDKey,
+        text: duration ? new HiddenString(`/location live ${duration}`) : new HiddenString('/location'),
+      })
+    )
+  }
+  // lifecycle
+  React.useEffect(() => {
+    const watchID = navigator.geolocation.watchPosition(
+      pos => {
+        dispatch(
+          Chat2Gen.createUpdateLastCoord({
+            coord: {
+              accuracy: Math.floor(pos.coords.accuracy),
+              lat: pos.coords.latitude,
+              lon: pos.coords.longitude,
+            },
+          })
+        )
+      },
+      () => {},
+      {enableHighAccuracy: isIOS, maximumAge: 0, timeout: 30000}
+    )
+    return () => {
+      navigator.geolocation.clearWatch(watchID)
+    }
+  }, [])
```

Comment:
This along with the `location` above can be extracted into a `useCurrentLocation` hook so it's clearer on the first look what's going on.